File: python/1974.minimum-time-to-type-word-using-special-typewriter.py
class Solution:
    def minTimeToType(self, word: str) -> int:
        res = 0
        prev = 'a'
        for letter in word:
            pos1 = min( ord(letter), ord(prev))
            pos2 = max( ord(letter), ord(prev))
            delta = pos2-pos1
            res += min(pos2-pos1, 26-pos2+pos1)
            prev = letter
        res += len(word)
        return res

File: cpp/1974.minimum-time-to-type-word-using-special-typewriter.cpp
class Solution {
public:
    int minTimeToType(string word) {
        int res = 0;
	char prev = 'a';
	//  bza
	//  1,27,0
	//   a b c
	//   0 1 2
	//   * *
	//   delta = 2
	for(auto letter:word){
		int pos1 = min(prev - 'a', letter-'a');
		int pos2 = max(prev - 'a', letter-'a');
		int delta = pos2-pos1;
		res += min(delta, 26-delta);
		prev = letter;
	}
	res += word.size();
        return res;	
    }
};
